pennylane/ops/functions/bind_new_parameters.py

```diff
@@ -198,8 +198,11 @@ def bind_new_parameters_pow(op: Pow, params: Sequence[TensorLike]):
     return Pow(bind_new_parameters(op.base, params), op.scalar)
 
 
-@bind_new_parameters.register
-def bind_new_parameters_hamiltonian(op: qml.Hamiltonian, params: Sequence[TensorLike]):
+@bind_new_parameters.register(qml.ops.Hamiltonian)
+@bind_new_parameters.register(qml.ops.LinearCombination)
+def bind_new_parameters_hamiltonian(
+    op: Union[qml.ops.Hamiltonian, qml.ops.LinearCombination], params: Sequence[TensorLike]
+):
     new_H = qml.Hamiltonian(params, op.ops)
```

Comment:
Again, if the user is somehow using a `Hamiltonian` with new op math enabled, should we give them `Hamiltonian` or should we say nope you're getting `LinearCombination` here although you gave me a `Hamiltonian`?

Reply:
Perhaps instead this line should be
```suggestion
    new_H = type(op)(params, op.ops)
```

Reply:
I wouldnt focus on mixed cases. If there is an easy fix: sure, but main priority for me would be for things to work with new opmath enabled, and be backward compatible with all-legacy opmath as well.